dot/state/block.go

```diff
@@ -291,28 +249,74 @@ func (bs *BlockState) SetHeader(header *types.Header) error {
 	return nil
 }
 
-// SetBlock will add a block to the DB
-func (bs *BlockState) SetBlock(block *types.Block) error {
-	// Add the blockHeader to the DB
-	err := bs.SetHeader(block.Header)
+// GetBlockDataSimple will return BlockData (Header and Body) for a given hash
+func (bs *BlockState) GetBlockDataSimple(hash common.Hash) (*types.BlockData, error) {
+	result := new(types.BlockData)
+
+	data, err := bs.db.Get(blockDataKey(hash))
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	blockData := &types.BlockData{
-		Hash:   block.Header.Hash(),
-		Header: block.Header.AsOptional(),
-		Body:   block.Body.AsOptional(),
+	r := &bytes.Buffer{}
+	_, err = r.Write(data)
+	if err != nil {
+		return nil, err
+	}
+
+	err = result.Decode(r)
+	if err != nil {
+		return nil, err
 	}
-	return bs.SetBlockData(blockData)
+
+	return result, nil
 }
 
-// SetBlockData will set the block data using given hash and blockData into DB
-func (bs *BlockState) SetBlockData(blockData *types.BlockData) error {
+// GetBlockDataAllFields will return what GetBlockDataSimple plus all other optional fields
+func (bs *BlockState) GetBlockDataAllFields(hash common.Hash) (*types.BlockData, error) {
```

Comment:
we shouldn't have any BlockData related methods in the state anymore, at least not something for getting the entire block data. if you want to move the bitmap parsing to here, and return BlockData that's in the bitmap, then that is okay, but it's unnecessary to do gets for all the fields when we might not need them